### ISSUE TITLE
Adding Progress Reports to the Website

### ISSUE THREAD

**Issue by logandgsmith** [ufosc/club-website#70]
Would there be interest in hosting progress reports? I know in the past we've talked about project leads giving status reports to the club, but perhaps we should just publish them to the website instead! 

## Prerequisite Work 
This would be the heavier option for this system. We'd need to do a few things to implement it.
- [ ] Finish backend for user logins
- [ ] Finish the club database for Single Source of Truth
- [ ] Allow server admin to assign roles / Pull project leads from database

##Base Work
After we accomplish these, we could begin implementing the progress report system which would include the following base features.
- Allow Project Leads to upload or write reports on site
- Allow for editing on reports
- Links in the projects page to the most recent report and an archive of old reports
- GitHub profile picture of writer
- Formatted Report (Possibly mimic GitHub Markdown).

## Stretch Goals
After the base features are implemented, we will want to add (a lot!) more polish onto the reports. These are features that would be nice to have, but only after we have a stable base.
- Create a more advanced editor, possibly resembling that on GitHub
- Show recent project updates on the home page
- Allow for Project Leads to insert a photo to represent their project, or have a single photo at the top (e.g. cover photo).
- Alert Project Leads with bot to write a new report
- Send an automated Tweet and Discord message with the project update

### Option 2: Pull reports from GitHub
This option would involve less writing and changes on the website, but it's also a less interesting option over time. For this, we would use the GitHub API to pull a report.md file (which project leads would write) from the project repos. This would get rid of the need to create an editor on our site, and fast-track getting the page up. If enough people want this feature, we can discuss it, but Option 1 seems like a much better option to me.

**Comment by swiftlee** [comment]
I think this has a good point to it. I think the easiest way to implement this is to link with the Discord bot and allow for project leads to push their progress report.

Having an editor would be something to talk about, but it just depends on how many people prefer one over the other.
